aws/table_aws_ec2_spot_price.go

```diff
@@ -0,0 +1,125 @@
+package aws
+
+import (
+	"context"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/turbot/steampipe-plugin-sdk/v4/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsEc2SpotPrice(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_ec2_spot_price",
+		Description: "AWS EC2 Spot Price History",
+		List: &plugin.ListConfig{
+			Hydrate: listEc2SpotPrice,
+			IgnoreConfig: &plugin.IgnoreConfig{
+				ShouldIgnoreErrorFunc: isNotFoundErrorV2([]string{"InvalidParameterValue"}),
+			},
+			KeyColumns: []*plugin.KeyColumn{
+				{Name: "availability_zone", Require: plugin.Optional},
+				{Name: "instance_type", Require: plugin.Optional},
+				{Name: "product_description", Require: plugin.Optional},
+				{Name: "start_time", Require: plugin.Optional},
+				{Name: "end_time", Require: plugin.Optional},
+			},
+		},
+		GetMatrixItemFunc: BuildRegionList,
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{Name: "availability_zone", Description: "The Availability Zone.", Type: proto.ColumnType_STRING},
+			{Name: "instance_type", Description: "The instance type.", Type: proto.ColumnType_STRING},
+			{Name: "product_description", Description: "A general description of the AMI.", Type: proto.ColumnType_STRING},
+			{Name: "spot_price", Description: "The maximum price per unit hour that you are willing to pay for a Spot Instance.", Type: proto.ColumnType_STRING},
+			{Name: "create_timestamp", Description: "The time stamp of the Spot price history.", Type: proto.ColumnType_TIMESTAMP, Transform: transform.FromField("Timestamp")},
```

Comment:
```suggestion
			{Name: "create_timestamp", Description: "The date and time the request was created, in UTC format.", Type: proto.ColumnType_TIMESTAMP, Transform: transform.FromField("Timestamp")},
```

I got this description from https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_SpotPrice.html, it seemed like you were using the other API property descriptions already.

Reply:
Also, is this the timestamp of when the spot price was set in the past, or is this the timestamp of when a user makes a request to retrieve that particular price?

Reply:
It represents the timestamp when the spot price was set in the past. Date filtering is done with `start_time` and `end_time` (which are not in the filters attribute :laughing:)

For example the following query : 
``` sql
select
    availability_zone,
    instance_type,
    product_description,
    spot_price::numeric as spot_price,
    start_time,
    end_time,
    create_timestamp as d1
from
    m6cloud_prod.aws_ec2_spot_price
where
        instance_type = 'm5.4xlarge'
  and product_description = 'Linux/UNIX'
  and availability_zone in
      (
       'eu-west-3a'
          )
  and start_time = now() - interval '1' month
  and end_time = now() - interval '1' minute
```

returns the following rows : 

![image](https://user-images.githubusercontent.com/107035817/201729604-8548384f-9fdd-44fc-a0f9-892ebf905414.png)